lib/smart_proxy_ansible/runner/ansible_runner.rb

```diff
@@ -168,14 +168,8 @@ def handle_broadcast_data(event)
 
       def write_inventory
         path = File.join(@root, 'inventory', 'hosts')
-        data_path = File.join(@root, 'data')
-        inventory_script = <<~INVENTORY_SCRIPT
-          #!/bin/sh
-          cat #{::Shellwords.escape data_path}
-        INVENTORY_SCRIPT
-        File.write(path, inventory_script)
-        File.write(data_path, JSON.dump(@inventory))
-        File.chmod(0o0755, path)
+        File.write(path, JSON.dump(@inventory))
+        File.chmod(0o0640, path)
```

Comment:
Doesn't this technically have a very short window where it could be world readable?

Reply:
I like the first approach and it looks like it's also correct. With the following script:
```ruby
#!/usr/bin/env ruby
File.open('/tmp/bla-test', 'w') do |file|
  file.chmod(0600)
  file.write('bla2')
end
```

A strace tells us the following syscalls are made:
```
openat(AT_FDCWD, "/tmp/bla-test", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 5
ioctl(5, TCGETS, 0x7ffdb80fabf0)        = -1 ENOTTY (Inappropriate ioctl for device)
fchmod(5, 0600)                         = 0
write(5, "bla2", 4)                     = 4
close(5)                                = 0
```

That order is correct.